packages/Toggle/src/Toggle.js

```diff
@@ -1,7 +1,354 @@
+import { colors } from '@leafygreen-ui/theme';
+
+const { css, cx } = emotion;
+
+const toggleInput = createDataProp('toggle-input');
+const toggleGroove = createDataProp('toggle-groove');
+
+const transitionInMS = 150;
+
+const inputStyle = css`
+  margin: 0;
+  position: absolute;
+  left: 100%;
+  top: 100%;
+  pointer-events: none;
+  opacity: 0;
+`;
+
+// We use a div for the focus state rather than a pseudo-element
+// because said pseudo-element would need to be on the label element
+// which can't use the contained input's focus pseudo-class.
+const focusStateStyle = css`
+  transition: all ${transitionInMS}ms ease-in-out;
+  position: absolute;
+  top: -2px;
+  bottom: -2px;
+  left: -2px;
+  right: -2px;
+  border: 2px solid rgba(67, 177, 229, 0.25);
+  border-radius: 50px;
+  opacity: 0;
+  transform: scale(0.8);
+
+  ${toggleInput.selector}:focus ~ & {
+    opacity: 1;
+    transform: scale(1);
+  }
+`;
+
+const labelStyleBase = css`
+  transition: all ${transitionInMS}ms ease-in-out;
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  margin: auto;
+  height: 11px;
+  line-height: 11px;
+  text-transform: uppercase;
+  font-weight: bold;
+  font-size: 11px;
+  color: white;
+  user-select: none;
+`;
+
+const onLabelStyle = css`
+  left: 9px;
+  color: #bbebff;
+
+  ${toggleInput.selector}:hover ~ ${toggleGroove.selector} > &,
+  ${toggleInput.selector}:focus ~ ${toggleGroove.selector} > & {
+    color: white;
+  }
+`;
+
+const offLabelStyle = css`
+  right: 6px;
+  color: #9fa1a2;
+`;
+
+const getStatefulContainerStyles = ({ size, disabled }) => {
+  const baseStyle = `
+    position: relative;
+    display: inline-block;
+    cursor: ${disabled ? 'not-allowed' : 'pointer'};
+  `;
+
+  const sizeStyle = {
+    default: `
+      height: 32px;
+      width: 62px;
+    `,
+
+    small: `
+      height: 22px;
+      width: 40px;
+    `,
+
+    xsmall: `
+      height: 14px;
+      width: 26px;
+    `,
+  };
+
+  return css`
+    ${baseStyle} ${sizeStyle[size]}
+  `;
+};
+
+const getStatefulGrooveStyles = ({ variant, checked, disabled }) => {
+  let baseStyle = `
+    transition: ${transitionInMS}ms all ease-in-out, 0 background-color linear;
+    display: inline-block;
+    overflow: hidden;
+    flex-shrink: 0;
+    position: relative;
+    border-radius: 50px;
+    overflow: hidden;
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    border: 1px solid;
+
+    // We're animating this pseudo-element in order to give the toggle groove
+    // background an animation in and out.
+    &:before {
+      content: '';
+      transition: ${transitionInMS}ms all ease-in-out;
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      border-radius: 50px;
+      background-color: ${colors.mongodb.blue};
+      box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.1);
+      opacity: 0;
+      transform: scale(0.85);
+    }
+  `;
+
+  if (disabled) {
+    baseStyle += `
```

Comment:
I don't love the string concatenation, but I can't think of a better alternative. We could use object syntax, but that gets super awkward when we deal with pseudo selectors - although maybe it's not that bad.

Reply:
I think I prefer concatenation over writing styles as objects still. Part of that might be my resentment of camelCase CSS though, and I definitely don't wanna have to write and read styles as:
```CSS
const styles = {
  'background-color': colors.green,
}
```
to have them be kebab-case.

Reply:
Yeah, I agree.